dotnet/notebooks/06-vector-stores-and-embeddings.ipynb

```diff
@@ -0,0 +1,464 @@
+    "## Define main components\n",
+    "\n",
+    "As soon as you define your data model using either attributes or record definition approach, you can start working with it using your database of choice. \n",
+    "\n",
+    "There are a couple of abstractions that allow you to work with your database and collections:\n",
+    "\n",
+    "1. `IVectorStoreRecordCollection<TKey, TRecord>` - represents a collection. This collection may or may not exist, and the interface provides methods to check if the collection exists, create it or delete it. The interface also provides methods to upsert, get and delete records. Finally, the interface inherits from `IVectorizedSearch<TRecord>` providing vector search capabilities.\n",
+    "2. `IVectorStore` - contains operations that spans across all collections in the vector store, e.g. `ListCollectionNames`. It also provides the ability to get `IVectorStoreRecordCollection<TKey, TRecord>` instances.\n",
+    "\n",
+    "Each connector has extension methods to register your vector store and collection using DI - `services.AddInMemoryVectorStore()` or `services.AddInMemoryVectorStoreRecordCollection(\"collection-name\")`. \n",
+    "\n",
+    "It's also possible to initialize these instances directly, which we are going to do in this notebook for simplicity:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {
+    "dotnet_interactive": {
+     "language": "csharp"
+    },
+    "polyglot_notebook": {
+     "kernelName": "csharp"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "using Microsoft.SemanticKernel.Connectors.InMemory;\n",
+    "\n",
+    "#pragma warning disable SKEXP0020\n",
+    "\n",
+    "// Define vector store\n",
+    "var vectorStore = new InMemoryVectorStore();\n",
+    "\n",
+    "// Get a collection instance using vector store\n",
+    "var collection = vectorStore.GetCollection<ulong, Glossary>(\"skglossary\");\n",
+    "\n",
+    "// Get a collection instance by initializing it directly\n",
+    "var collection2 = new InMemoryVectorStoreRecordCollection<ulong, Glossary>(\"skglossary\");"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Initializing a collection instance will allow you to work with your collection and data, but it doesn't mean that this collection already exists in a database. To ensure you are working with existing collection, you can create it if it doesn't exist:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "dotnet_interactive": {
+     "language": "csharp"
+    },
+    "polyglot_notebook": {
+     "kernelName": "csharp"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "await collection.CreateCollectionIfNotExistsAsync();"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, since we just created a new collection, it is empty, so we want to insert some records using a data model we defined above:"
```

Comment:
```suggestion
    "Now, since we just created a new collection, it is empty, so we want to insert some records using the data model we defined above:"
```